components/EmotionForm.js

```diff
@@ -2,79 +2,43 @@ import { emotions } from "@/lib/emotions";
 import { useState } from "react";
 import styled from "styled-components";
 
-export default function EmotionForm({ onCreateEmotion }) {
+export default function EmotionForm({ onSubmit, defaultValue }) {
   const currentDateTime = new Date(
     new Date().getTime() - new Date().getTimezoneOffset() * 60000
   )
     .toISOString()
     .slice(0, 16);
 
-  const [selectedEmotion, setSelectedEmotion] = useState("");
-  const [selectedIntensity, setSelectedIntensity] = useState(5);
-  const [selectedDateTime, setSelectedDateTime] = useState(currentDateTime);
   const [formError, setFormError] = useState(""); // Error message if not all fields are filled out on submit
+  const hasError = formError !== "";
   const [successMessage, setSuccessMessage] = useState(""); // Success message
-  const [hasError, setHasError] = useState(false); // New state variable for styling
-
-  // Change emotion
-  function handleEmotionChange(event) {
-    setSelectedEmotion(event.target.value);
-    if (event.target.value !== "") {
-      setFormError(""); // Reset error message when the field is filled
-      setHasError(false); // Reset styling once an emotion is selected
-    }
-  }
-
-  // Change intensity
-  function handleIntensityChange(event) {
-    setSelectedIntensity(event.target.value);
-  }
-
-  // Change dateTime
-  function handleDateTimeChange(event) {
-    setSelectedDateTime(event.target.value);
-    if (event.target.value) {
-      setFormError(""); // Reset error message when date/time is selected
-    }
-  }
+  const [selectedIntensity, setSelectedIntensity] = useState(
```

Comment:
You use this `useState` as `defaultValue` for the `input`-field with the `type="range"`, right? With the other form fields you always write the value directly as `defaultValue`, e.g. with `select`: 
```defaultValue={defaultValue?.emotion || ""}```

Why don't you do that here, is there a reason for it? 🤔  Otherwise I would recommend doing the same here and defining the value directly as `defaultValue` for the range-input-field.

Reply:
Thank you for your hint and we tested it: When the user wants to edit the intensity, the intensity value would still be default value although the slider shows a different position. We testet your suggestion but would need a useEffect and do not want to restructure it.